### DISCUSSION TITLE
Servo on Redox

### DISCUSSION THREAD

**Discussion by eshikafe** [servo/servo#27696] [General]
Hello, Please has anyone tried to run servo on redox os?

**Comment by jdm** [comment]
By default it would require redox support in https://github.com/rust-windowing/winit/ and https://github.com/servo/surfman. I'm not sure if Redox has any support for GL or EGL yet, which is a requirement.

**Reply by eshikafe** [reply]
Good. Looks feasible. I am actually thinking of a browser OS concept based on Servo, WebAssembly and Redox. The entire UI/Apps will be rendered with Servo and optimized with WebAssembly.

**Comment by Thatoo** [comment]
@eshikafe any news on your project?

**Reply by eshikafe** [reply]
Hello @Thatoo. I am yet to finalize the best architecture for the OS.

Some open points: 
1. Replacing [Orbital](https://gitlab.redox-os.org/redox-os/orbital) with a WebGPU based windowing system.
2. Replacing `syscalls` with `WASI`.
3. What javascript engine will be best suited?
4. How to achieve milli-seconds start-up time for the browserOS?
5. How to get all Electron apps to run on this browserOS by default.

Are you interested in the browserOS project? 

Thanks.

**Reply by SamuraiCrow** [reply]
If you have a WebAssembly JIT or AOT compiler, have you considered using Javy from the BytecodeAlliance? It is a JavaScript to WebAssembly converter with a WinterCG compliant runtime in the works (designed to compete with NodeJS, I assume) but is written in Rust. This is in response to point 3.

**Comment by webbeef** [comment]
Hi @eshikafe , where can I read more about the browserOS project? Looks quite interesting to me!

**Reply by Thatoo** [reply]
Same question.

**Reply by eshikafe** [reply]
I am yet to setup a page for the project. 
I will need to call the OS another name - `browserOS` doesn't sound like an appropriate name based on what I have in mind. The OS itself uses web technologies for everything userspace.

**Reply by anoraktrend** [reply]
> I am yet to setup a page for the project. I will need to call the OS another name - `browserOS` doesn't sound like an appropriate name based on what I have in mind. The OS itself uses web technologies for everything userspace.

WebOS is taken, Maybe SpiderOS?

**Reply by spiritofstar** [reply]
maybe OxideOS,looks neat to me and you should use Boa JS engine written in rust but it is not fully complete but mostly complete.

**Comment by DavidYummy** [comment]
W rizzlet in Ohio Rizz Rizz me up rizzx
